### ISSUE TITLE
Freqtrade is not buying after upgrade to 2021.7

### ISSUE THREAD

**Issue by iggyfrog** [freqtrade/freqtrade#5403]
hi.  After upgrade to 2021.7 I see that the bot is not getting the correct ask-strategy.  In your changelog you say that you moved some configuration to root config.

2 bots one on version develop-6763bd44 (1) and one on develop-f17942b6 (2).

the second does not buy anything.  
on show config from Telegram I get 
1
Ask strategy: 
{"use_sell_signal": true, "sell_profit_only": false, "ignore_roi_if_buy_signal": false, "sell_profit_offset": 0.0, "ignore_buying_expired_candle_after": 0, "price_side": "ask"}
2
Ask strategy: 
{"price_side": "ask"}

Can you please help what exactly I need to migrate to make it work again?

**Comment by xmatthias** [comment]
While there have been some changes to settings (where they  should be located) in the last version - this will not cause any issues as it will only add some "deprecation" warnings to the log, but everything will continue to work as is.

If a setting is incompatible and MUST be changed, the bot will fail to start (to avoid having "inactive" bots running forever).

If a bot is not buying, this is usually caused by the strategy. 
You should also carefully check the logs, maybe there's an issue with API keys or similar.

**Comment by aristosv** [comment]
I guess the question here is: Older freqtrade versions configuration would use these options for the ask strategy

```
    "ask_strategy":{
        "use_sell_signal": true,
        "sell_profit_only": false,
        "ignore_roi_if_buy_signal": false,
        "sell_profit_offset": 0.0,
        "ignore_buying_expired_candle_after": 0,
        "price_side": "ask"
   }
```

Whereas now, the [full example of the configuration](https://github.com/freqtrade/freqtrade/blob/4115121c249077ff123c2d15ac22df62ed375ed4/config_examples/config_full.example.json#L43) only lists these options

```
    "ask_strategy":{
        "price_side": "ask",
        "use_order_book": true,
        "order_book_top": 1
    },
```

Would that affect the bot's decision on when (or how frequently) to buy?

**Comment by xmatthias** [comment]
no, this will not affect buying (never has) - but selling.

However - there is compatibility code that is transitioning attributes like `sell_profit_offset` (...) from `ask_strategy` to the root level unless it's defined in both (in which case, it would stop the bot during startup).

**Comment by aristosv** [comment]
ok, I will try to provide as much useful information as I can because there's definitely something going on here, but I can't figure out what.

Bot1 is running Freqtrade develop-f920c268 and it's using [this ](https://paste.ofcode.org/dPyJxMDRke39wVw33D9cBJ) configuration and [this ](https://paste.ofcode.org/NxS2UcMMnVS5bLNGwQ8Kdr)strategy

Bot2 is running Freqtrade develop-4115121c and it's using [this ](https://paste.ofcode.org/r5gUgKQnj6d9vQwfsQfu7g) configuration and [this ](https://paste.ofcode.org/jX2mJZWefvfdFbHcyRBXAj)strategy

I have been tailing the log files for both bots for hours and there are no errors. But Bot1 is trading crypto, and Bot2 hasn't opened a single trade for 2 days now.

The strategy is exactly the same, except for the protections in Bot2 that have been moved to the strategy file instead of the configuration file. The configuration files have some minor differences but I can't figure out if those differences are causing the issue with Bot2 not trading.

**Comment by xmatthias** [comment]
I have my doubts that the bot that's trading can work with the shown strategy.
Most likely, it's running a slightly different strategy (maybe one with the same name in a different file).

The strategies have a "static" sell signal that's always on (no condition). 
``` python
    def populate_sell_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        dataframe.loc[
            (),
            'sell'] = 1
        return dataframe
```

This will prevent any buys - as it would result in immediate sells (the sell-signal is there).
This is however not a change that's been made or introduced lately, but it's a a fact that's been there since "forever" (at least 3 years).

**Comment by aristosv** [comment]
Yesterday I changed the strategy to this one, just to see if it would make a difference.

```
from freqtrade.strategy.interface import IStrategy
from typing import Dict, List
from functools import reduce
from pandas import DataFrame
import talib.abstract as ta
import freqtrade.vendor.qtpylib.indicators as qtpylib
class freqtrade_BTC_aristos(IStrategy):

    protections = [
        {
            "method": "StoplossGuard",
            "lookback_period": 1440,     # in the last 24 hours
            "trade_limit": 5,            # if 5 trades hit stoploss
            "stop_duration": 60,         # stop trading for an hour
            "only_per_pair": "false"     # applies to all pairs
        },
        {
            "method": "CooldownPeriod",
            "stop_duration": 5           # after closing a trade wait 5 minutes before re-buying the same pair
        },
        {
            "method": "MaxDrawdown",
            "lookback_period": 360,      # in the last 6 hours
            "trade_limit": 20,           # considering only pairs that made at least 20 trades
            "max_allowed_drawdown": 0.2, # if drawdown is more than 20%
            "stop_duration": 60          # stop all trading for 60 minutes
        },
        {
            "method": "LowProfitPairs",
            "lookback_period": 360,      # in the last 6 hours
            "trade_limit": 2,            # considering only pairs that made at least 2 trades
            "required_profit": 0.02,     # with less than 2% profit
            "stop_duration": 60          # stop trading the specific pairs for 60 minutes
        }
    ]

    stoploss = -0.3
    timeframe = '1h'
    trailing_stop_positive = 0.01
    trailing_stop_positive_offset = 0.02
    trailing_only_offset_is_reached = True

    def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        macd = ta.MACD(dataframe)
        dataframe['macd'] = macd['macd']
        dataframe['macdsignal'] = macd['macdsignal']
        dataframe['macdhist'] = macd['macdhist']
        dataframe['cci'] = ta.CCI(dataframe)
        return dataframe

    def populate_buy_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        dataframe.loc[
            (
                qtpylib.crossed_above(dataframe['macd'], dataframe['macdsignal']) &
                (dataframe['cci'] <= -50.0)
            ),
            'buy'] = 1
        return dataframe

    def populate_sell_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        dataframe.loc[
            (
                qtpylib.crossed_below(dataframe['macd'], dataframe['macdsignal']) &
                (dataframe['cci'] >= 100.0)
            ),
            'sell'] = 1
        return dataframe
```

Still no luck though. The bot doesn't buy There are no errors in the log file. The bot is not trying and failing. [This ](https://paste.ofcode.org/PF7xxtG5nGdSmtn6g4HYR)is the log file for the last couple of days.

**Comment by PaulFinch** [comment]
Same behaviour here, default strategy (ex: Strategy005) with brand new config from `freqtrade new-config --config config.json`. No trading at all (only in dry mode).

**Comment by aristosv** [comment]
If I can provide any additional information that can help pinpoint the issue, I can do that.

**Comment by xmatthias** [comment]
I'd not know how to help you further to be honest.
Freqtrade is buying as always for me - so it's probably "something" on your end.

What exactly - i don't know. Maybe the log shows something - but ohter than that i can't tell you (maybe the markets changed so your strategy is currently not buying).

**Comment by pampos79** [comment]
I have the same problem. A strategy works well with Dry Run but the same strategy with live run is not opening any trades. I am using VULTR with a Germany Based server. I also whitelisted the server ip in Binance.

**Comment by xmatthias** [comment]
The most likely reason (and almost certainly only reason) for this problem (assuming you're sure the same strategy works on bots from earlier in 2021) is #4382.
This means: you're either using the pandas-native `.rolling().std()` or `qtpylib.bollinger_bands()` in combination with BTC pairs.

The pandas-bug caused std to truncate output if the values are very low (BTC pairs are usually priced with 0.0000x or lower) - causing std to be 0.

Depending how your strategy is using the output of the `.std()` calculation, this can obviously cause missing buy signals. In the end however, it means that the condition that used this is invalid and based on a pandas bug that's now been fixed.

There's nothing we can do to bring it back - and i highly doubt the validity / sustainability of strategies that were based on the occurance of this bug.

The updating fix was released by pandas early july (Pandas  1.3.0) and merged early in July - which means it'll be included in 2021.7.
